snapshotter/demux/proxy/snapshotter.go

```diff
@@ -39,5 +64,10 @@ func NewProxySnapshotter(ctx context.Context, address string, dialer func(contex
 	if err != nil {
 		return nil, err
 	}
-	return proxy.NewSnapshotter(snapshotsapi.NewSnapshotsClient(gRPCConn), address), nil
+
+	if metricsProxy != nil {
+		go metricsProxy.Serve(ctx)
```

Comment:
Possibly for followup PR:

Should we have something smarter than just thowing a goroutine into the wild? (e.g. somewhere that selects over an error channel so we can be notified if the metrics proxy unexpectedly exits?)

Reply:
yea definitely should manage this goroutine explicitly over a channel... I've created issue #607 to track it and added a TODO comment for follow up PR to this one.